scripts/entity/components/movement_components/player_movement_component.gd

```diff
@@ -116,6 +146,11 @@ func _moth_jump(delta: float) -> void:
 		super.apply_gravity(delta)
 	
 	
+func apply_climbing_movement(delta: float) -> void:
+	velocity.y = move_toward(velocity.y, _current_input_direction.y * CLIMB_SPEED, ACCELERATION * delta)
+	velocity.x = move_toward(velocity.x, _current_input_direction.x * MAX_SPEED, ACCELERATION * delta)
```

Comment:
I wouldnt worry about this right now but I think a few of the movement applications need to be in the base movement